fixtures/prov_filter.py

```diff
@@ -1,61 +1,48 @@
-from utils.conf import cfme_data
 from utils.log import logger
-
-
-def provider_keys():
-    return cfme_data.get('management_systems', {}).keys()
-
-
-class ProviderFilter(object):
-    def __init__(self, defaults):
-        self._filtered_providers = defaults
-
-    @property
-    def providers(self):
-        return self._filtered_providers
-
-    @providers.setter
-    def providers(self, filtered):
-        self._filtered_providers = parse_filter(filtered)
-
-    def __contains__(self, provider):
-        return provider in self.providers
-
-filtered = ProviderFilter(provider_keys())
+from utils.providers import (
+    ProviderFilter, list_providers, global_filters, get_crud
+)
 
 
 def pytest_addoption(parser):
     # Create the cfme option group for use in other plugins
     parser.getgroup('cfme')
     parser.addoption("--use-provider", action="append", default=[],
-        help="list of providers or tags to include in test")
+        help="list of provider keys or provider tags to include in test")
 
 
 def pytest_configure(config):
-    """ Filters the list of providers as part of pytest configuration. """
+    """ Filters the list of providers as part of pytest configuration
+
+    Note:
+        Additional filter is added to the global_filters dict of active filters here.
+    """
 
     cmd_filter = config.getvalueorskip('use_provider')
     if not cmd_filter:
         cmd_filter = ["default"]
 
-    filtered.providers = cmd_filter
+    new_filter = parse_filter(cmd_filter)
+    global_filters['use_provider'] = new_filter
 
     logger.debug('Filtering providers with {}, leaves {}'.format(
-        cmd_filter, filtered.providers))
+        cmd_filter, [p.key for p in list_providers()]))
 
 
 def parse_filter(cmd_filter):
     """ Parse a list of command line filters and return a filtered set of providers.
 
     Args:
         cmd_filter: A list of ``--use-provider`` options.
-    """
 
-    filtered_providers = provider_keys()
-    for provider in provider_keys():
-        data = cfme_data['management_systems'][provider]
-        tags = data.get('tags', [])
-        if ('disabled' in tags) or \
-           ((provider not in cmd_filter) and not (set(tags) & set(cmd_filter))):
-            filtered_providers.remove(provider)
-    return filtered_providers
+    Returns:
+        A :py:class:`utils.providers.ProviderFilter`.
+    """
+    try:
+        # It's a key!
+        get_crud(cmd_filter)
+        pf = ProviderFilter(keys=[cmd_filter])
+    except KeyError:
+        # It's a list of tags!
+        pf = ProviderFilter(required_tags=cmd_filter)
```

Comment:
So, the old method, iirc, allowed you to mix providers and tags on a single line, this looks like it's an either or?